pkg/machines/components/desktopConsole.jsx

```diff
@@ -0,0 +1,219 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from "react";
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+const MoreInformationInstallVariant = ({ os, command, innerHtml }) => {
+    return (
+        <li className='machines-desktop-install-instructs-item'>
+            <div className='machines-desktop-install-instructs-row'>
+                <b>{os}:</b>&nbsp;
+                {innerHtml && <div dangerouslySetInnerHTML={{__html: innerHtml}}/>}
+                {!innerHtml && <div className='machines-desktop-shell-command'>{command}</div>}
+            </div>
+        </li>
+
+    );
+}
+
+const MoreInformationContent = () => {
+    const msg1 = cockpit.format(_("Clicking \"Launch Remote Viewer\" will download a .vv file and launch $0."),
+        '<i>Remote Viewer</i>');
+
+    const msg2 = cockpit.format(_("$0 is available for most operating systems. To install it, search for it in GNOME Software or run the following:"),
+        '<i>Remote Viewer</i>');
+
+    const downloadMsg = cockpit.format(_("Download the MSI from $0"),
+        '<a href="https://virt-manager.org/download/" target="_blank">virt-manager.org</a>');
+
+    return (
+        <div>
+            <br />
+            <p className='machines-desktop-more-info-text' dangerouslySetInnerHTML={{__html: msg1}} />
+            <p className='machines-desktop-more-info-text' dangerouslySetInnerHTML={{__html: msg2}} />
+
+            <ul className='machines-desktop-install-instructs'>
+                <MoreInformationInstallVariant os='RHEL, CentOS' command='sudo yum install virt-viewer'/>
+                <MoreInformationInstallVariant os='Fedora' command='sudo dnf install virt-viewer'/>
+                <MoreInformationInstallVariant os='Ubuntu, Debian' command='sudo apt-get install virt-viewer'/>
+                <MoreInformationInstallVariant os='Windows' innerHtml={downloadMsg}/>
```

Comment:
It's not a big deal, if you prefer it that way then let's just keep it as it is.

Thanks for the other fixes! I'll re-review tomorrow morning.

Is there something that can be done about http:// not working with the inline viewer? Such as not offering it in the first place if SSL is off, instead of showing a broken widget?